openapi-comparator-tests/src/test/scala/sttp/apispec/openapi/validation/OpenAPIComparatorTest.scala

```diff
@@ -0,0 +1,395 @@
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    val missMatchIssue = MissMatch("style")
+    val headerIssue = IncompatibleHeader("X-Rate-Limit", List(missMatchIssue))
+    val responsesIssue = IncompatibleResponse(List(headerIssue))
+    val operationIssue = IncompatibleOperation("get", List(responsesIssue))
+    val pathIssue = IncompatiblePath("/pets/{petId}", List(operationIssue))
+    val expected = List(pathIssue)
+
+    assert(openAPIComparator.compare() == expected)
+  }
+
+  test("incompatible header explode") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_2_2.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_2_1.json").flatMap(_.as[OpenAPI]): @unchecked
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    val missMatchIssue = MissMatch("explode")
+    val headerIssue = IncompatibleHeader("X-Rate-Limit", List(missMatchIssue))
+    val responsesIssue = IncompatibleResponse(List(headerIssue))
+    val operationIssue = IncompatibleOperation("get", List(responsesIssue))
+    val pathIssue = IncompatiblePath("/pets/{petId}", List(operationIssue))
+    val expected = List(pathIssue)
+
+    assert(openAPIComparator.compare() == expected)
+  }
+
+  test("incompatible header allowEmptyValue") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_2_3.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_2_2.json").flatMap(_.as[OpenAPI]): @unchecked
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    val missMatchIssue = MissMatch("allowEmptyValue")
+    val headerIssue = IncompatibleHeader("X-Rate-Limit", List(missMatchIssue))
+    val responsesIssue = IncompatibleResponse(List(headerIssue))
+    val operationIssue = IncompatibleOperation("get", List(responsesIssue))
+    val pathIssue = IncompatiblePath("/pets/{petId}", List(operationIssue))
+    val expected = List(pathIssue)
+
+    assert(openAPIComparator.compare() == expected)
+  }
+
+  test("incompatible header allowReserved") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_2_4.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_2_3.json").flatMap(_.as[OpenAPI]): @unchecked
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    val missMatchIssue = MissMatch("allowReserved")
+    val headerIssue = IncompatibleHeader("X-Rate-Limit", List(missMatchIssue))
+    val responsesIssue = IncompatibleResponse(List(headerIssue))
+    val operationIssue = IncompatibleOperation("get", List(responsesIssue))
+    val pathIssue = IncompatiblePath("/pets/{petId}", List(operationIssue))
+    val expected = List(pathIssue)
+
+    assert(openAPIComparator.compare() == expected)
+  }
+}
```

Comment:
Apart from the naming issues, I think what I'm missing is positive test cases (different schemas, but compatible).

Another thing: evolution of the request body, e.g. adding a new variant to a coproduct. Both in the request, and response bodies. Depending where the variant is added (client/server and response/request), we'll get compatible, or incompatible changes

Reply:
Another missing test case: changing metadata - adding or removing examples, description itd - this should not influence compatibility

Reply:
Yet another: adding an alternative media type to the response. E.g. there's an application/json, but now the server can also return XML. Or the client can send XML.

Reply:
And one more: introducing new response codes. E.g. 200 was defined, now additionally 400 and 401 are defined

Reply:
Is changing the parameter's name a compatible change? E.g. was `/pet/{petId}`, now is: `/pet/{id}`?

Reply:
ok, so we should test this as well